[c10] Allow taking a std::tuple as arg

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #32938 [WIP][quantization] Use torchbind for Linear PackedParams
* #32977 [JIT] Resolve custom classes in source importer
* **#32948 [c10] Allow taking a std::tuple as arg**

Differential Revision: [D19709119](https://our.internmc.facebook.com/intern/diff/D19709119)

Commit:
[c10] Allow taking a std::tuple as arg

[ghstack-poisoned]

Commit:
Update on "[c10] Allow taking a std::tuple as arg"


Differential Revision: [D19709119](https://our.internmc.facebook.com/intern/diff/D19709119)

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit 19ca3a9f:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=19ca3a9f276b29478ed73d5b445d3c821412e471).

### :detective: 1 new failure recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4407737) pytorch_windows_test2 (1/1)
**Step:** "Test" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=94842408) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=94842408))
<details>
<summary>
<code>RuntimeError: test_cpp_extensions failed!</code>
</summary>

```
---------------------------------------------------------------------- 
Ran 32 tests in 460.218s 
 
FAILED (errors=2, skipped=1) 
Not wiping extensions build folder because Windows 
Traceback (most recent call last): 
  File "run_test.py", line 457, in <module> 
    main() 
  File "run_test.py", line 450, in main 
    raise RuntimeError(message) 
RuntimeError: test_cpp_extensions failed! 
 
(base) circleci@PACKER-5E29F737 C:\Users\circleci\project\test>if ERRORLEVEL 1 exit /b 1  
+ cleanup
+ retcode=1
+ set +x
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

Commit:
Update on "[c10] Allow taking a std::tuple as arg"


Differential Revision: [D19709119](https://our.internmc.facebook.com/intern/diff/D19709119)

[ghstack-poisoned]

Commit:
Update on "[c10] Allow taking a std::tuple as arg"


Differential Revision: [D19709119](https://our.internmc.facebook.com/intern/diff/D19709119)

[ghstack-poisoned]

Commit:
Update on "[c10] Allow taking a std::tuple as arg"


Differential Revision: [D19709119](https://our.internmc.facebook.com/intern/diff/D19709119)

[ghstack-poisoned]

Commit:
Update on "[c10] Allow taking a std::tuple as arg"


Differential Revision: [D19709119](https://our.internmc.facebook.com/intern/diff/D19709119)

[ghstack-poisoned]

Commit:
Update on "[c10] Allow taking a std::tuple as arg"


Differential Revision: [D19709119](https://our.internmc.facebook.com/intern/diff/D19709119)

[ghstack-poisoned]

Commit:
Update on "[c10] Allow taking a std::tuple as arg"


Differential Revision: [D19709119](https://our.internmc.facebook.com/intern/diff/D19709119)

[ghstack-poisoned]

Commit:
Update on "[c10] Allow taking a std::tuple as arg"


Differential Revision: [D19709119](https://our.internmc.facebook.com/intern/diff/D19709119)

[ghstack-poisoned]

Commit:
Update on "[c10] Allow taking a std::tuple as arg"


Differential Revision: [D19709119](https://our.internmc.facebook.com/intern/diff/D19709119)

[ghstack-poisoned]

Commit:
Update on "[c10] Allow taking a std::tuple as arg"


Differential Revision: [D19709119](https://our.internmc.facebook.com/intern/diff/D19709119)

[ghstack-poisoned]

Commit:
Update on "[c10] Allow taking a std::tuple as arg"


Differential Revision: [D19709119](https://our.internmc.facebook.com/intern/diff/D19709119)

[ghstack-poisoned]

Commit:
Update on "[c10] Allow taking a std::tuple as arg"


Differential Revision: [D19709119](https://our.internmc.facebook.com/intern/diff/D19709119)

[ghstack-poisoned]

driazati reviewed (approved):
lgtm

Commit:
Update on "[c10] Allow taking a std::tuple as arg"


Differential Revision: [D19709119](https://our.internmc.facebook.com/intern/diff/D19709119)

[ghstack-poisoned]
